### ISSUE TITLE
pg_catalog - SELECT on pg_proc returns only a subset for a user with DQL on only pg_catalog

### ISSUE THREAD

**Issue by proddata** [crate/crate#11481]
**CrateDB version**:
4.5.1

**Environment description**:
local / tarball

**Problem description**:

For users with limited privileges (e.g. only on a specific schema **and** `pg_catalog`), CrateDB only returns a subset of all records for `pg_proc`. Only granting `DQL` on all schemas allows access to all records. This leads to a problem with PowerBI, as one of the intial queries fails (see https://community.crate.io/t/powerbi-desktop-npgsql-error/722/4 )

### no privileges 

```sql
-- with user crate
CONNECT OK
cr> CREATE USER testuser WITH (password='testuser');                                                                                                                                                                           
CREATE OK, 1 row affected  (0.161 sec)
```

user with non privileges:
```sql
-- with user testuser
cr> SELECT * FROM pg_proc;                                                                                                                                                                                                      
SchemaUnknownException[Schema 'pg_catalog' unknown]
```

---

### DQL on pg_catalog
```sql
-- with user crate
cr> GRANT DQL on SCHEMA pg_catalog TO testuser;
GRANT OK, 1 row affected  (0.088 sec)
```
      
user with non privileges:
```sql
-- with user testuser
cr> SELECT COUNT(*) FROM pg_proc;                                                                                                                                                                                               
+----------+
| count(*) |
+----------+
|       24 |
+----------+
SELECT 1 row in set (0.001 sec)
```

---

### DQL on all schemas

```sql
-- with user crate
cr> GRANT DQL TO testuser;
GRANT OK, 1 row affected  (0.088 sec)
```

user with DQL privileges on all schemas
```sql
-- with user testuser
cr> SELECT COUNT(*) FROM pg_proc;                                                                                                                                                                                               
+----------+
| count(*) |
+----------+
|      544 |
+----------+
SELECT 1 row in set (0.001 sec)
```

**Steps to reproduce**:

**Comment by proddata** [comment]
This _might_ be fixed with https://github.com/crate/crate/commit/760b4054d6e406f1f0668f294693d123b9ba1835 @jeeminso

**Comment by proddata** [comment]
Just tried it with the nightly 4.6 docker image, and the SELECT on `pg_proc` returns the full table, so basically this can be closed, unless we plan to backport it to 4.5.x

**Comment by jeeminso** [comment]
> This _might_ be fixed with [760b405](https://github.com/crate/crate/commit/760b4054d6e406f1f0668f294693d123b9ba1835) @jeeminso

Yes, just to confirm, it is fixed by the change mentioned.
